### PR TITLE
[WIP] Add `wsk sdk install bashauto` command to generate CLI auto completion bash script

### LINKED ISSUE #802: go CLI loses bash tab completion available with python CLI
The python CLI offers tab completion in bash (see [here](https://github.com/openwhisk/openwhisk/blob/master/tools/ubuntu-setup/bashprofile.sh)) which is referenced in the documentation [here](https://github.com/openwhisk/openwhisk/tree/master/tools/vagrant/custom#use-the-wsk-cli-inside-the-vm) (we used to have more than this reference @csantanapr):

    **Tip:** Login into the VM using `vagrant ssh` then use the `wsk` CLI with tab completion on commands and parameters.

The go CLI loses tab completion.

@dubeejw @mdeuser

### PR DESCRIPTION
**Pull request by mdeuser**
Add command to generate bash auto-completion script
Fixes #802

### CODE REVIEW

**Comment by mdeuser** [comment]
@dubeejw - Ready for review

**Comment by mdeuser** [comment]
PG 349

File: tools/go-cli/go-whisk-cli/commands/sdk.go [hunk 1]
```diff
@@ -72,9 +73,18 @@ var sdkInstallCmd = &cobra.Command{
...
             err = iOSInstall()
+        case "bashauto":
+            err = WskCmd.GenBashCompletionFile(BASH_AUTOCOMPLETE_FILENAME)
+            if (err != nil) {
+                whisk.Debug(whisk.DbgError, "GenBashCompletionFile('%s`) error: \n", BASH_AUTOCOMPLETE_FILENAME, err)
+                errStr := fmt.Sprintf("Unable to generate '%s': %s", BASH_AUTOCOMPLETE_FILENAME, err)
+                werr := whisk.MakeWskError(errors.New(errStr), whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG, whisk.NO_DISPLAY_USAGE)
+                return werr
+            }
+            fmt.Printf("ok. To enable command line completion of wsk commands, source the auto completion script (%s) into your bash environment\n", BASH_AUTOCOMPLETE_FILENAME)
```
Comment (dubeejw) [inline tools/go-cli/go-whisk-cli/commands/sdk.go]: Perhaps change "ok" to "Successfully installed Bash auto-completion," or remove the "ok" altogether.

File: tools/go-cli/go-whisk-cli/commands/sdk.go [hunk 2]
```diff
@@ -60,7 +61,7 @@ var sdkInstallCmd = &cobra.Command{
         var err error
         if len(args) != 1 {
             whisk.Debug(whisk.DbgError, "Invalid number of arguments: %d\n", len(args))
-            errStr := fmt.Sprintf("The SDK component argument is invalid.  One component (docker, swift, or ios) must be specified")
+            errStr := fmt.Sprintf("The SDK component argument is missing.  One component (docker, swift, ios, or bashauto) must be specified")
```
Comment (dubeejw) [inline tools/go-cli/go-whisk-cli/commands/sdk.go]: Two spaces between sentences?

File: tools/go-cli/go-whisk-cli/commands/sdk.go [hunk 3]
```diff
@@ -72,9 +73,18 @@ var sdkInstallCmd = &cobra.Command{
...
+                whisk.Debug(whisk.DbgError, "GenBashCompletionFile('%s`) error: \n", BASH_AUTOCOMPLETE_FILENAME, err)
+                errStr := fmt.Sprintf("Unable to generate '%s': %s", BASH_AUTOCOMPLETE_FILENAME, err)
+                werr := whisk.MakeWskError(errors.New(errStr), whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG, whisk.NO_DISPLAY_USAGE)
+                return werr
+            }
+            fmt.Printf("ok. To enable command line completion of wsk commands, source the auto completion script (%s) into your bash environment\n", BASH_AUTOCOMPLETE_FILENAME)
         default:
             whisk.Debug(whisk.DbgError, "Invalid component argument '%s'\n", component)
-            errStr := fmt.Sprintf("The SDK component argument '%s' is invalid.  Valid components are docker, swift, and ios", component)
+            errStr := fmt.Sprintf("The SDK component argument '%s' is invalid.  Valid components are docker, swift, ios, and bashauto", component)
```
Comment (dubeejw) [inline tools/go-cli/go-whisk-cli/commands/sdk.go]: Two spaces between sentences?

Comment (rabbah) [inline tools/go-cli/go-whisk-cli/commands/sdk.go]: > Two spaces between sentences?

:) some of us are older than others.

Comment (mdeuser) [inline tools/go-cli/go-whisk-cli/commands/sdk.go]: The debate rages on :-).
https://en.wikipedia.org/wiki/Sentence_spacing

**Comment by mdeuser** [comment]
Just now seeing the above valid comment regarding the test case.  Alas, the PR is already merged; I'll open another for this test update.

**Pull request merged**